pkg/frontend/frontend_label_names.go

```diff
@@ -4,13 +4,39 @@ import (
 	"context"
 
 	"github.com/bufbuild/connect-go"
+	"github.com/grafana/dskit/tenant"
+	"github.com/prometheus/common/model"
 
 	"github.com/grafana/pyroscope/api/gen/proto/go/querier/v1/querierv1connect"
 	typesv1 "github.com/grafana/pyroscope/api/gen/proto/go/types/v1"
+	"github.com/grafana/pyroscope/pkg/util"
 	"github.com/grafana/pyroscope/pkg/util/connectgrpc"
+	"github.com/grafana/pyroscope/pkg/validation"
 )
 
 func (f *Frontend) LabelNames(ctx context.Context, c *connect.Request[typesv1.LabelNamesRequest]) (*connect.Response[typesv1.LabelNamesResponse], error) {
 	ctx = connectgrpc.WithProcedure(ctx, querierv1connect.QuerierServiceLabelNamesProcedure)
+
+	tenantIDs, err := tenant.TenantIDs(ctx)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	if util.HasTimeRange(c.Msg) {
```

Comment:
I guess it should return ok, interval and be named model.RequestTimeRange() ?

Reply:
Done in https://github.com/grafana/pyroscope/pull/2498/commits/0d5f92cc2cd98ac9b9e81080ac882e59026b1ae7.